src/v/iceberg/tests/values_json_test.cc

```diff
@@ -223,7 +223,7 @@ static const std::vector<values_json_test_case> test_cases{
     "decimal_value (trailing decimal point)",
     decimal_value{absl::int128{std::numeric_limits<long>::max()}},
     decimal_type{21, 0},
-    R"("9223372036854775807.")",
+    R"("9223372036854775807")",
```

Comment:
em? the test explicitly states it tests with trailing decimal

other tests too, they explicitly don't have integral parts